<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>801</width>
    <height>570</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="ViewWidget" name="viewWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>227</y>
      <width>783</width>
      <height>291</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>781</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string>Control</string>
    </property>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="geometry">
      <rect>
       <x>490</x>
       <y>20</y>
       <width>111</width>
       <height>144</height>
      </rect>
     </property>
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QSplitter" name="splitter_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Point size</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="PointsizespinBox">
         <property name="value">
          <number>2</number>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QSplitter" name="splitter_5">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Centroid size</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="CentroidspinBox">
         <property name="value">
          <number>15</number>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>20</y>
       <width>161</width>
       <height>144</height>
      </rect>
     </property>
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QSplitter" name="splitter_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Number of iterations</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="IterationspinBox">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QSplitter" name="splitter_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Frame rate</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="FramespinBox">
         <property name="value">
          <number>1</number>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>20</y>
       <width>171</width>
       <height>143</height>
      </rect>
     </property>
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QSplitter" name="splitter_8">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Kmeans initialization</string>
         </property>
        </widget>
        <widget class="QComboBox" name="methodcomboBox">
         <item>
          <property name="text">
           <string>random_real</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>random_sample</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>kmeans_plusplus</string>
          </property>
         </item>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QSplitter" name="splitter">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Number of clusters</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="ClusterspinBox">
         <property name="value">
          <number>2</number>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QGroupBox" name="groupBox">
     <property name="geometry">
      <rect>
       <x>14</x>
       <y>20</y>
       <width>141</width>
       <height>142</height>
      </rect>
     </property>
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QSplitter" name="splitter_7">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>t-SNE dimension</string>
         </property>
        </widget>
        <widget class="QComboBox" name="dimension_tsne_comboBox">
         <property name="editable">
          <bool>false</bool>
         </property>
         <item>
          <property name="text">
           <string>2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>3</string>
          </property>
         </item>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QSplitter" name="splitter_6">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>File name</string>
         </property>
        </widget>
        <widget class="QComboBox" name="filenamecomboBox">
         <property name="currentText">
          <string>a.txt</string>
         </property>
         <item>
          <property name="text">
           <string>a.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>b.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>c.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>d.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>a_3d.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>b_3d.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>c_3d.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>d_3d.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>MNIST_test100.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>MNIST_test1000.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>MNIST_test3000.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>MNIST_test4000.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>MNIST_test5000.txt</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>MNIST_test.txt</string>
          </property>
         </item>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QSlider" name="rotationSlider">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>170</y>
       <width>160</width>
       <height>16</height>
      </rect>
     </property>
     <property name="value">
      <number>50</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="zoomSlider">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>200</y>
       <width>160</width>
       <height>16</height>
      </rect>
     </property>
     <property name="value">
      <number>50</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>170</y>
       <width>60</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotation</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>200</y>
       <width>60</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Zoom</string>
     </property>
    </widget>
    <widget class="QSlider" name="translate_x_Slider">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>170</y>
       <width>160</width>
       <height>16</height>
      </rect>
     </property>
     <property name="value">
      <number>50</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="translate_y_Slider">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>200</y>
       <width>160</width>
       <height>16</height>
      </rect>
     </property>
     <property name="value">
      <number>50</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>170</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Translation x</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>200</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Translation y</string>
     </property>
    </widget>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>620</x>
       <y>40</y>
       <width>111</width>
       <height>120</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Run</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stop_pushButton">
        <property name="text">
         <string>Stop/Start</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="next_pushButton">
        <property name="text">
         <string>Next</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="prev_pushButton">
        <property name="text">
         <string>Prev</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>801</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <addaction name="menuView"/>
   <addaction name="menuEdit"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionActual_Size">
   <property name="text">
    <string>Actual Size</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+0</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="text">
    <string>Zoom Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+2</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ViewWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>ViewWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <slots>
  <slot>zoomActualSize()</slot>
  <slot>zoomIn()</slot>
  <slot>zoomOut()</slot>
 </slots>
</ui>
